main.go

```diff
@@ -84,7 +84,7 @@ func main() {
 			os.Exit(1)
 		}
 	}
-
+	pipelinesv1alpha5.DefaultProvider = ctrlConfig.Spec.DefaultProvider
```

Comment:
worth putting a doc here to explain why it has to be set like this

Reply:
still no inline doc here, would agree it would be useful, maybe a `TODO:` remove for v1alpha7 release?

Reply:
Added todo in https://github.com/sky-uk/kfp-operator/pull/394/commits/97c408776a235828d74af923cad516b52bbb4c20